src/main/scala/com.snowplowanalytics.s3/loader/IBufferedOutput.scala

```diff
@@ -0,0 +1,74 @@
+package com.snowplowanalytics.s3.loader
+
+import com.snowplowanalytics.s3.loader.model.S3Config
+import com.snowplowanalytics.s3.loader.serializers.ISerializer
+import org.joda.time.{DateTime, DateTimeZone}
+import org.joda.time.format.DateTimeFormat
+import org.slf4j.Logger
+
+import scala.collection.JavaConversions._
+
+trait IBufferedOutput {
+  val log: Logger
+  val serializer: ISerializer
+  val s3Emitter: S3Emitter
+  val s3Config: S3Config
+
+  def getBaseFilename(startTime: Long, endTime: Long): String = {
```

Comment:
It would be extremely useful to make this and other `private` fields `private[this]` and move them down below the public API for the `IBufferedOutput`.

Reply:
Hey there @peel, sorry but I didn't' get that.
I am new to scala too, I guess I need more descriptive comments.
I understood that I should make this `private[this]` and move it out of `IBufferedOutput`, is it right? I tried to make the `val`s in the beginning to be private but it seems that I cannot do that on traits.

Reply:
You did very well on that one! This kind of contributions really makes you speed into new languages and helps Snowplow community along the way. 
My remark about moving was mostly based on principle that it's a good practice (and AFAIR part of the Scala Style Guide) to keep public members (like methods, functions, fields) on top of the file whereas private should in general live in the bottom as they are merely implementation details. It simplifies understanding on what's the interface like. 
So for this particular trait I'd organize it in something like:
```
trait IBufferedOutput {
  val log: Logger
  val serializer: ISerializer
  val s3Emitter: S3Emitter
  val s3Config: S3Config

  def flushMessages(): Unit = _

  private[this] def getBaseFilename(startTime: Long, endTime: Long) = _

  private[this] val TimeFormat = _
  private[this] val DateFormat = _

  private[this] val folderDateFormat = _
  private[this] def formatFolderDatePrefix(currentTime: DateTime) = _
```

Hope this helps. And once again thanks for contributing!

Reply:
Thank for your comments!
I will send a commit shortly.